tuplex/python/tuplex/dataset.py

```diff
@@ -109,22 +112,49 @@ def collect(self):
         assert self._dataSet is not None, 'internal API error, datasets must be created via context objects'
         return self._dataSet.collect()
 
-    def take(self, nrows=5):
-        """ action that generates a physical plan, processes data and collects the top results then as list of tuples.
+    def take(self, limitTop=5, limitBottom=0):
+        """ action that generates a physical plan, processes data and collects the top and bottom results
+        then as list of tuples.
 
         Args:
-            nrows (int): number of rows to collect. Per default ``5``.
+            limitTop (int): number of top rows to collect. Per default ``5``.
+            limitBottom (int): number of bottom rows to collect. Per default ``0``.
         Returns:
             (list): A list of tuples
 
         """
+        assert limitTop is None or isinstance(limitTop, int), 'num rows must be an integer or None'
+        assert limitBottom is None or isinstance(limitBottom, int), 'num bottom last must be an integer or None'
 
-        assert isinstance(nrows, int), 'num rows must be an integer'
-        assert nrows > 0, 'please specify a number greater than zero'
+        if limitTop is None or limitTop < 0:
+            limitTop = -1
+
+        if limitBottom is None or limitBottom < 0:
+            limitBottom = -1
 
         assert self._dataSet is not None, 'internal API error, datasets must be created via context objects'
 
-        return self._dataSet.take(nrows)
+        return self._dataSet.take(limitTop, limitBottom)
+
+    def head(self, nrows):
```

Comment:
we need some end-to-end tests (python tests) for this functionality. I.e., add them in python/tests.